src/libraries/System.Drawing.Common/src/System/Drawing/Icon.Windows.COMWrappers.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Drawing.Internal;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+
+namespace System.Drawing
+{
+    public sealed partial class Icon : MarshalByRefObject, ICloneable, IDisposable, ISerializable
+    {
+        public unsafe void Save(Stream outputStream)
+        {
+            if (_iconData != null)
+            {
+                outputStream.Write(_iconData, 0, _iconData.Length);
+            }
+            else
+            {
+                if (outputStream == null)
+                    throw new ArgumentNullException(nameof(outputStream));
+
+                // Ideally, we would pick apart the icon using
+                // GetIconInfo, and then pull the individual bitmaps out,
+                // converting them to DIBS and saving them into the file.
+                // But, in the interest of simplicity, we just call to
+                // OLE to do it for us.
+                PICTDESC pictdesc = PICTDESC.CreateIconPICTDESC(Handle);
+                Guid g = DrawingComWrappers.IPicture.Guid;
+                IntPtr lpPicture;
+                DrawingComWrappers.CheckStatus(OleCreatePictureIndirect(&pictdesc, &g, false, &lpPicture));
+
+                IntPtr streamPtr = IntPtr.Zero;
+                try
+                {
+                    DrawingComWrappers.IPicture picture = (DrawingComWrappers.IPicture)DrawingComWrappers.Instance.GetOrCreateObjectForComInstance(lpPicture, CreateObjectFlags.None);
+
+                    var gpStream = new GPStream(outputStream, makeSeekable: false);
+                    streamPtr = DrawingComWrappers.Instance.GetOrCreateComInterfaceForObject(gpStream, CreateComInterfaceFlags.None);
+
+                    DrawingComWrappers.CheckStatus(picture.SaveAsFile(streamPtr, -1, out _));
+                }
+                finally
+                {
+                    if (streamPtr != IntPtr.Zero)
```

Comment:
This gets back to the `IDisposable` discussion regarding `Marshal.ReleaseComObject()`. I would suggest we agree on some common `ComWrappers` convention in the BCL around how to manually release wrapper instances. An example of a possible convention can be observed in the [C#/WinRT](https://github.com/microsoft/CsWinRT/blob/master/src/WinRT.Runtime/ObjectReference.cs) repo.

Reply:
I've added `IDisposable` to `IPicture`, and implement it in the `PictureWrapper` class now. Let me know what you think of the new iteration.